tests/DocumentationExamplesTest.php

```diff
@@ -1575,6 +1575,103 @@ public function testCausalConsistency(): void
         ob_end_clean();
     }
 
+    public function testSnapshotQueries(): void
```

Comment:
I observed SnapshotUnavailable(246) errors in CI builds. This PR will require work-arounds for [replica sets](https://github.com/mongodb/mongo-ruby-driver/commit/7c4117b58e3d12e237f7536f7521e18fc15f79ac#diff-7b02c2b4b34a58c688750366d0230014bb453e0381af394d736aa9d0a25538a5R374) and [sharded clusters](https://github.com/mongodb/mongo-ruby-driver/commit/7c4117b58e3d12e237f7536f7521e18fc15f79ac#diff-3f7d999c8369e37a728bdeb5eff37a5780d98403c143e6e4b64008d174e1bcdaR10), which we can adapt from [RUBY-2909](https://jira.mongodb.org/browse/RUBY-2909).

Reply:
Added private `waitForSnapshot()` and `preventStaleDbVersionError()` methods for replica sets and sharded clusters, respectively.